src/models/concrete_syntax.rs

```diff
@@ -82,20 +85,70 @@ pub(crate) fn get_all_matches_for_concrete_syntax(
   (matches, !is_empty)
 }
 
-/// `find_next_sibling` navigates the cursor through the tree to the next sibling.
-/// If no sibling exists, it ascends the tree until it can move laterally or until it reaches the root.
+fn find_next_sibling(cursor: &mut TreeCursor) -> bool {
+  while !cursor.goto_next_sibling() {
+    if !cursor.goto_parent() {
+      return false;
+    }
+  }
+  true
+}
+
+/// Attempts to match a given ConcreteSyntax template against a sequence of sibling nodes
+/// in an Abstract Syntax Tree (AST).
 ///
 /// # Arguments
 ///
-/// * `cursor` - A mutable reference to a `TreeCursor` used to navigate the tree.
+/// * `cursor` - A mutable reference to the TreeCursor, which is used to navigate the AST.
+/// * `source_code` - A slice of bytes representing the source code being analyzed.
+/// * `meta` - A reference to the ConcreteSyntax template used for matching.
 ///
-/// The function mutates the cursor position. If no further siblings exist, the cursor ends at the root.
-fn find_next_sibling(cursor: &mut TreeCursor) {
-  while !cursor.goto_next_sibling() {
-    if !cursor.goto_parent() {
-      break;
-    }
+/// # Returns
+///
+/// A tuple containing:
+/// * A HashMap where keys are variable names from the template and values are CapturedNode instances representing matched AST nodes.
+/// * A boolean indicating whether a match was found.
+/// * An Option containing the range of matched nodes if a match was found.
+///
+/// # Algorithm
+///
+/// 1. Initialize cursor to the first child and iterate through siblings.
+/// 2. Use `get_matches_for_node` to attempt matching the template against the subtree starting at each sibling.
+/// 3. If a match is found, determine the range of matched nodes and return the match mapping, status, and range.
+/// 4. If no match is found, return an empty mapping, false status, and None for range.
+pub(crate) fn match_sequential_siblings(
+  cursor: &mut TreeCursor, source_code: &[u8], meta: &ConcreteSyntax,
+) -> (HashMap<String, CapturedNode>, bool, Option<Range>) {
+  let the_node = cursor.node();
+  let mut child_incr = 0;
+  let node_str = the_node.utf8_text(source_code).unwrap();
+
+  if cursor.goto_first_child() {
+    // Iterate through siblings to find a match
+    while {
+      let mut tmp_cursor = cursor.clone();
+      let (mapping, mut matched, indx) =
+        get_matches_for_node(&mut tmp_cursor, source_code, meta, true, the_node);
+
+      if matched {
+        // Determine the last matched node. Remember, we are matching subsequences of children [n ... k]
+        let mut last_node = the_node.child(the_node.child_count() - 1);
+        if let Some(last_node_index) = indx {
+          last_node = the_node.child(last_node_index);
+          matched = matched && (last_node_index != child_incr || the_node.child_count() == 1);
```

Comment:
A bit convoluted and should be refactored.

Here we say, we match if:

1. we were able to align the sequence with the templates
2. the sequence is of length > 1, UNLESS the node only has one child